core/src/bytes/zarith/zarith.h

```diff
@@ -0,0 +1,164 @@
+                std::reverse(data.begin(), data.end());
+
+                std::vector<uint8_t> result;
+                size_t id = 0, delay = 0; // delay is used when offset > 7
+                uint8_t offset = 0;
+                while (id - delay < data.size()) {
+                    auto byte = data[id - delay];
+
+                    // We shift current byte
+                    byte <<= offset;
+
+                    // Take shifted bits from previous byte
+                    byte |= id - delay > 0 ? data[id - delay - 1] >> (7 - (offset - 1)) : 0x00;
+
+                    // This boolean will let us know if shifting last byte of inputData
+                    // will generate a new byte (with remaining shifted bytes)
+                    bool isLast = data[id - delay] == data.back();
+                    bool needAdditionalByte = isLast && data[id - delay] >> (7 - offset);
+
+                    // We set first bit to 1 to know that there is a coming byte
+                    if (!isLast || needAdditionalByte) {
+                        byte |= 0x80;
+                    }
+
+                    // Push to result modified byte
+                    result.push_back(byte);
+                    if (needAdditionalByte) {
+                        result.push_back(data[id - delay] >> (7 - offset));
+                    }
+
+                    // Update offset: if offset % 7 == 0 then following byte
+                    // is not shifted i.e. offset = 0
+                    offset = offset > 0 && !(offset % 7) ? 0 : offset + 1;
+
+                    // If we shifted with [id/7] > 7 then we have a "delay"
+                    // meaning that we should read values from "delayed" byte
+                    delay = id / 7;
+
+                    id++;
+                }
+                return result;
+            };
+
+            static std::vector<uint8_t> zParse(const std::vector<uint8_t> &inputData, bool forceContinue = false) {
+                // Reverse because result corresponds to Little Endian
+                BytesReader bytesReader(inputData);
+                return zParse(bytesReader);
+            };
+
+            // The mode where we force parser to continue parsing even when finding
+            // a null bit was made just for testing purpose
+            static  std::vector<uint8_t> zParse(BytesReader &reader, bool forceContinue = false) {
+                // We get bytes to parse without altering
+                // bytes reader's state
+                auto cursor = reader.getCursor();
+                std::vector<uint8_t> data(reader.readUntilEnd());
+                reader.reset();
+                reader.read(cursor);
+
+                std::vector<uint8_t> result;
+                size_t id = 0, delay = 0;
+                uint8_t offset = 0;
+                while (id + delay < data.size()) {
+                    auto byte = data[id + delay];
+                    auto isLast = byte == data.back();
```

Comment:
Yes I agree !

Reply:
You didn't fix this.

Reply:
Alright, I see what happened here, I fixed it but in the wrong file ...  Navigation on my IDE was taking me to `cmake-build-debug/include/ledger/core/bytes/zarith/zarith.h` instead of `core/bytes/zarith/zarith.h` ... :trollface: